numpy/core/src/multiarray/nditer_constr.c

```diff
@@ -1132,6 +1132,9 @@ npyiter_prepare_one_operand(PyArrayObject **op,
 
                 /* Replace with a new descr which is in native byte order */
                 nbo_dtype = PyArray_DescrNewByteorder(*op_dtype, NPY_NATIVE);
+                if (nbo_dtype == NULL) {
+                    return 0;
+                }
                 Py_DECREF(*op_dtype);
                 *op_dtype = nbo_dtype;
```

Comment:
Can't suggest a full change unfortunately, I would consider removing the `nbo_dtype` completely:
```
Py_SETREF(*op_dtype, PyArray_DescrNewByteorder(*op_dtype, NPY_NATIVE));
if (*op_dtype == NULL) {
    return 0;
}
```
simpler, and currently I think `*op_dtype` may be missing (depends on whether the caller expects cleanup on error).

Reply:
done